app/routes/events_.$eventId.tsx

```diff
@@ -29,13 +29,27 @@ export async function loader({ params }: LoaderFunctionArgs) {
 
 export async function action({ params, request }: ActionFunctionArgs) {
   const userSession = await requireUserSession(request);
+  const formData = await request.formData();
+  const intent = formData.get('intent');
 
-  const event = await db.event.update({
-    where: { id: params.eventId },
-    data: { users: { connect: { id: userSession.userId } } },
-  });
-
-  return json({ event });
+  switch (intent) {
+    case 'joinEvent': {
+      await db.event.update({
+        where: { id: params.eventId },
+        data: { users: { connect: { id: userSession.userId } } },
+      });
+      return redirect(`/events/${params.eventId}`);
+    }
+    case 'deleteEvent': {
+      await db.event.delete({
```

Comment:
in the future, we should probably make the delete action, just get marked as deleted, but not actually removed from the database.

so when participants re-visit the page it wouldn't be a 404